src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -170,14 +171,19 @@ def __enter__(self):
                 self._flow_create_kwargs,
                 **common_kwargs,
             )
+            self._processes_manager.start_processes()
+            # If the spawned process that created the fork process has not started successfully,
+            # raise 'SpawnedForkProcessManagerStartFailure' exception and without continue execution.
+            if not self._processes_manager._is_spawned_fork_process_manager_healthy:
+                ex = SpawnedForkProcessManagerStartFailure()
+                raise ex
         else:
             executor_creation_func = partial(FlowExecutor.create, **self._flow_create_kwargs)
             # 1. Create input_queue, output_queue, and _process_info in the main process.
             # 2. Spawn _n_process sub-process and pass the above queue/dict to these sub-process to transfer information
             # between main process and sub process.
             self._processes_manager = SpawnProcessManager(executor_creation_func, **common_kwargs)
-
-        self._processes_manager.start_processes()
+            self._processes_manager.start_processes()
```

Comment:
I think we can have a method in the base class `ensure_healthy()`, the the logic here is common:
```
self._process_manager.start_processes()
self._process_manager.ensure_healthy()
```

Then then we can check and raise the exception in the process manager implementation.
And we can also have some validation logic for spawn mode.